Assets/Scripts/Game/TalkManager.cs

```diff
@@ -1922,16 +1946,27 @@ public void RestoreConversationData(SaveDataConversation data)
 
                         string key = item.Symbol.Name;
 
-                        if (questInfo.resourceInfo.ContainsKey(key)) // if orphaned list of quest resources contains a matching entry
+                        if (questInfo.resourceInfo.ContainsKey(key)) // if list of quest resources contains a matching entry
                         {
                             questInfo.resourceInfo[key].questResource = item; // update (relink) it
                         }
                     }
 
                 }
             }
-
             listRumorMill = data.listRumorMill;
```

Comment:
I've found an old save (player is on main quest) where data.listRumorMill returns null. This causes the subsequent for loop to throw an exception. It might be worth doing some null handling on this line as well. I can provide copy of save if needed. :)